starter-code/src/functions-and-arrays.js

```diff
@@ -11,13 +18,46 @@ var words = [
   'crackpot'
 ];
 
-// Calculating a Sum
+function findLongestWord(){
+var longitud = 0;
+var mPalabra = "";
+
+for (var i = 0; i < words.length; i++) {
+    if (words[i].length > longitud) {
+        longitud = words[i].length;
+        mPalabra = words[i];
+    
+    }else words[i].length==0
+    return(undefined)
+}
+return (mPalabra);
+}
+
 
+// Calculating a Sum
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray(array)
+{
+ // var sumatoria = array.reduce(function(a, b){
+  //return a + b; //Regresa el acumulador más el siguiente
+//}, 0); //Pero si no encuentras nada o no hay siguiente, regresa 0
+//var promedio = sumatoria / array.length;
+var reducer = (accumulator, currentValue)=> accumulator + currentValue;
+console.log(array.reduce(reducer));
+}
+
+sumArray(numbers)
+
 // Calculate the Average
 
 var numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+function averageNumbers(array)
+{
+var reducer = (accumulator, currentValue)=> accumulator + currentValue;
+ console.log(array.reduce(reducer)/array.length);
+}
+averageNumbers(numbersAvg)
 
```

Comment:
Falta la la parte de calcular la media de las palabras, una solución puede ser esta 
function averageWordLength(arr){
  if(arr.length==0){return undefined}
  var suma= 0
  for(var k=0; k<words.length; k++){
    suma += words[k].length;
  }
  if(words.length !==0){
    return parseInt(suma/words.length);
  }
}